web/src/reducers/budget.js

```diff
@@ -497,6 +497,89 @@ const buildBudget = bigState => {
     });
   });
 
+  // Get the first activity called "Program Administration" and use it as the
+  // basis for computing quarterly federal share.  Then reudce that to an
+  // object whose keys are FFYs and whose values are an array of percentages
+  // for fiscal quarters.
+  const percentPerQuarter = Object.entries(
+    Object.values(bigState.activities.byKey).filter(
+      a => a.name === 'Program Administration'
+    )[0].quarterlyFFP
+  ).reduce(
+    (percentYear, [ffy, quarters]) => ({
+      ...percentYear,
+      [ffy]: Object.entries(quarters)
+        .sort(([q1], [q2]) => +q1 - +q2)
+        .map(([, q]) => q.state / 100)
+    }),
+    {}
+  );
+
+  /**
+   * Update a budget section with key personnel cost numbers
+   * @param {Object} section The section to be updated; should include
+   *                 top-level properties corresponding to fiscal years along
+   *                 with a total.  Those objects have federal, state, and
+   *                 total properties.
+   * @param {Number} year The federal fiscal year to update
+   * @param {Number} federal Federal share of the cost to add
+   * @param {Number} state State share of the cost to add
+   * @param {Number} total Total cost to add
+   */
+  const updateSectionWithKP = (section, year, federal, state, total) => {
+    section[year].federal += federal;
+    section[year].state += state;
+    section[year].total += total;
+    section.total.federal += federal;
+    section.total.state += state;
+    section.total.total += total;
+  };
+
+  // We also have to account for the cost attributable to key personnel.  This
+  // isn't added to any activity, but instead gets rolled up into the totals.
+  // It is always a 90% federal/10% state split.  We're just going to make sure
+  // we only look at key personnel for whom the "has costs" flag is set, just
+  // to be safe.
+  keyPersonnel
+    .filter(kp => kp.hasCosts)
+    .forEach(kp => {
+      Object.entries(kp.costs).forEach(([year, cost]) => {
+        // Get the state and federal share.  Always 90/10 (for now).
```

Comment:
This is definitely a question for @NAretakis or @jeromeleecms - I think for HITECH it will always be 90/10, but even if so, we should almost certainly document that this is happening somewhere.  Maybe there should be a whole thing in the documentation that describe how all the budget stuff is calculated - like all the inter-relationships and stuff.

Instead of hard-coding this, we could also have it use the FFP level from the "Program Administration" activity, since that's also where it gets its quarterly distribution.  That one is an HIT activity, so it's always 90/10 too (though users are allowed to change it to 75/25 or 50/50 and maybe they shouldn't be?).